src/test/java/io/jenkins/plugins/analysis/core/quality/QualityGateTest.java

```diff
@@ -282,6 +285,261 @@ void shouldBeFailedWhenTotalFailedAndNewUnstableThresholdIsLowerThanWarningCount
+
+        QualityGateResult qualityGateResult = new QualityGateResult(totalThresholdResult, totalUnstableThresholdResult,
+                newFailedThresholdResult, newUnstableThresholdResult);
+
+        QualityGate qualityGate = this.createGateWithBuilder();
+
+        ArrayList<String> evaluations = (ArrayList<String>) qualityGateResult.getEvaluations(analysisResult,
+                qualityGate);
+        assertThat(evaluations).hasSize(4);
+
+        assertThat(evaluations.get(0)).contains("FAILURE");
+        assertThat(evaluations.get(1)).contains("UNSTABLE");
+        assertThat(evaluations.get(2)).contains("FAILURE");
+        assertThat(evaluations.get(3)).contains("UNSTABLE");
+    }
+
+    @Test
+    void shouldReturnEvaluationForTotalFailed() {
+        ThresholdResult totalThresholdResult = mock(ThresholdResult.class);
+        ThresholdResult totalUnstableThresholdResult = mock(ThresholdResult.class);
+        ThresholdResult newFailedThresholdResult = mock(ThresholdResult.class);
+        ThresholdResult newUnstableThresholdResult = mock(ThresholdResult.class);
+
+        when(totalThresholdResult.isTotalReached()).thenReturn(true);
+        when(totalThresholdResult.isHighReached()).thenReturn(true);
+        when(totalThresholdResult.isNormalReached()).thenReturn(true);
+        when(totalThresholdResult.isLowReached()).thenReturn(true);
+
+        AnalysisResult analysisResult = mock(AnalysisResult.class);
+        when(analysisResult.getTotalSize()).thenReturn(15);
+
+        QualityGateResult qualityGateResult = new QualityGateResult(totalThresholdResult, totalUnstableThresholdResult,
+                newFailedThresholdResult, newUnstableThresholdResult);
+
+        QualityGate qualityGate = this.createGateWithBuilder();
+
+        Collection<String> evaluations = qualityGateResult.getEvaluations(analysisResult, qualityGate);
+        assertThat(evaluations).hasSize(4);
+
+        testEvaluationMessageAllFailure(evaluations);
+    }
+
+    @Test
+    void shouldReturnEvaluationForNewFailed() {
+        ThresholdResult totalThresholdResult = mock(ThresholdResult.class);
+        ThresholdResult totalUnstableThresholdResult = mock(ThresholdResult.class);
+        ThresholdResult newFailedThresholdResult = mock(ThresholdResult.class);
+        ThresholdResult newUnstableThresholdResult = mock(ThresholdResult.class);
+
+        when(newFailedThresholdResult.isTotalReached()).thenReturn(true);
+        when(newFailedThresholdResult.isHighReached()).thenReturn(true);
+        when(newFailedThresholdResult.isNormalReached()).thenReturn(true);
+        when(newFailedThresholdResult.isLowReached()).thenReturn(true);
+
+        AnalysisResult analysisResult = mock(AnalysisResult.class);
+        when(analysisResult.getTotalSize()).thenReturn(10);
+
+        testEvaluation(totalThresholdResult, totalUnstableThresholdResult, newFailedThresholdResult,
+                newUnstableThresholdResult, analysisResult, false);
+    }
```

Comment:
Evtl. würde es Sinn machen, alle vier Methoden in eine zusammenzufassen und nur mit 2 Stubs zu arbeiten: `resultAllReachedFalse` und `resultAllReachedTrue`. Dann könnte man nach dem Setup nur die 4 Assertions hinhängen:
```
testEvaluation(resultAllReachedFalse, resultAllReachedFalse, resultAllReachedTrue,
               resultAllReachedFalse, analysisResult, FAILURE);
testEvaluation(resultAllReachedFalse, resultAllReachedTrue, resultAllReachedFalse,
               resultAllReachedFalse, analysisResult, FAILURE);
testEvaluation(resultAllReachedTrue, resultAllReachedFalse, resultAllReachedFalse,
               resultAllReachedFalse, analysisResult, UNSTABLE);
testEvaluation(resultAllReachedFalse, resultAllReachedFalse, resultAllReachedFalse,
               resultAllReachedTrue, analysisResult, UNSTABLE);
```